main.go

```diff
@@ -84,13 +85,26 @@ func main() {
 		LeaderElectionID:   "988b4062.3scale.net",
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		setupLog.Error(err, "unable init the manager")
+		os.Exit(1)
+	}
+
+	secretLabelSelector, err := apimachinerymetav1.ParseToLabelSelector(k8sutils.ApicastSecretLabel)
+	if err != nil {
+		setupLog.Error(err, "unable parse apicast secrets label")
+		os.Exit(1)
+	}
+
+	if secretLabelSelector == nil {
+		setupLog.Info("secretLabelSelector is empty")
 		os.Exit(1)
 	}
```

Comment:
Oh okay, but the function is currently hardcoded to take in a constant string `apimachinerymetav1.ParseToLabelSelector(k8sutils.ApicastSecretLabel)` so looking at the function itself, I don't think there would be a time where the returned `secretLabelSelector`  and `err` can both be nil :thinking: Also `apimachinerymetav1.ParseToLabelSelector()` looks to only take in a string value so I dont think it can take in a nil reference unless I'm missing something :thinking: 

Overall I don't feel strongly on this, no harm on doing the extra checking to not chance a nil pointer :+1: